TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/DrivebaseAuto.java

```diff
@@ -0,0 +1,91 @@
+import com.arcrobotics.ftclib.command.CommandScheduler;
+import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
+import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
+import com.acmerobotics.roadrunner.Action;
+import com.acmerobotics.roadrunner.Pose2d;
+import com.acmerobotics.roadrunner.SequentialAction;
+import com.acmerobotics.roadrunner.Vector2d;
+import com.acmerobotics.roadrunner.ftc.Actions;
+import org.firstinspires.ftc.teamcode.roadrunner.MecanumDrive;
+
+import org.firstinspires.ftc.teamcode.*;
+
+@Autonomous(name="Drive-Base-Auto", group = "Real")
+public class DrivebaseAuto extends LinearOpMode {
+
+    @Override
+    public void runOpMode() throws InterruptedException {
+
+        Robot robot = new Robot(this);
+
+        CommandScheduler.getInstance().reset();
+        CommandScheduler.getInstance().cancelAll();
+
+
+        int allianceNumber = 1;//blue = 1, red = -1
+
+        //tell the bot where it starts
+        Pose2d initialPose = new Pose2d(-3 * Constants.INCH_TO_TILE * allianceNumber, 1.5 * Constants.INCH_TO_TILE * allianceNumber, Math.toRadians(90 * allianceNumber));
+        MecanumDrive drive = new MecanumDrive(hardwareMap, initialPose);
+
+        //decide where we should park
+        boolean isGettingNeutral = true;
+        boolean isTouchingLowRung = false;
+
+        Pose2d updatedPose = new Pose2d(drive.pose.position.x,drive.pose.position.y,drive.pose.heading.toDouble());
+
+
+        TrajectoryActionBuilder getNeutral = drive.actionBuilder(initialPose)
+                .splineToConstantHeading(new Vector2d(-1 * Constants.INCH_TO_TILE * allianceNumber,2 * Constants.INCH_TO_TILE * allianceNumber),Math.toRadians(270 * allianceNumber))
+                .strafeTo(new Vector2d(-2.8 * Constants.INCH_TO_TILE * allianceNumber,2.8 * Constants.INCH_TO_TILE * allianceNumber));
+
+        TrajectoryActionBuilder touchLowRung = drive.actionBuilder(updatedPose)
+                .strafeTo(new Vector2d(-0.5 * Constants.INCH_TO_TILE*allianceNumber,1.5 * Constants.INCH_TO_TILE * allianceNumber))
+                .strafeTo(new Vector2d(-0.5 * Constants.INCH_TO_TILE*allianceNumber,0.5 * Constants.INCH_TO_TILE * allianceNumber));//trying to touch the low rung
+
+        TrajectoryActionBuilder parkObzone = drive.actionBuilder(updatedPose)
+                .strafeTo(new Vector2d(-2.5 * Constants.INCH_TO_TILE * allianceNumber,-2.5 * Constants.INCH_TO_TILE * allianceNumber));//drive straight into ob zone
+
+
+
+
+        waitForStart();
+
+        //to be tested
+        while (opModeIsActive() && !isStopRequested()) {
+            robot.run();
+            drive.updatePoseEstimate();//show where the bot think itself at
+
+            telemetry.addData("x", drive.pose.position.x);
+            telemetry.addData("y", drive.pose.position.y);
+            telemetry.addData("heading (deg)", Math.toDegrees(drive.pose.heading.toDouble()));
+            telemetry.update();
+        }
```

Comment:
I wish you would actually read my comments. Last time I said that all this cannot be placed after this while loop. The order in which code is written does matter; code is sequential - what is written before happens before.